src/transformers/benchmark/cache_benchmark.py

```diff
@@ -0,0 +1,180 @@
+
+    def target(self, batch_size, max_cache_length, seq_length, cache_type, mode):
+        self.model.generation_config.max_length = max_cache_length
+        input_ids = (
+            self.tokenizer([self.input_text] * batch_size, return_tensors="pt")
+            .to(self.device)
+            .input_ids[:batch_size, :seq_length]
+        )
+
+        self.model._reset_cache()
+
+        if cache_type == "static":
+            self.model._setup_cache(StaticCache, batch_size, max_cache_len=max_cache_length)
+
+        generated_ids = torch.zeros(batch_size, self.num_iter)
+
+        model = self.model
+        past_key_values = None
+
+        set_seed(0)
+
+        if cache_type == "dynamic":
+            past_key_values = DynamicCache()
+
+        if mode == "compiled":
+            torch.compiler.reset()
+            with torch.no_grad():
+                model = torch.compile(self.model, mode="reduce-overhead", fullgraph=True)
+
+        with torch.no_grad():
+            outputs = model(input_ids, past_key_values=past_key_values, return_dict=False, use_cache=True)
+            input_ids = sample(outputs[0], temperature=0.6, top_k=5)[0]
+            generated_ids[:, 0] = input_ids[:, 0]
+
+        _input_ids = input_ids
+
+        def target():
+            input_ids = _input_ids
+
+            torch.cuda.synchronize()
+
+            with torch.no_grad():
+                for i in range(1, self.num_iter):
+                    outputs = model(input_ids, past_key_values=past_key_values, return_dict=False, use_cache=True)
+                    input_ids = sample(outputs[0], temperature=0.6, top_k=5)[0]
+                    generated_ids[:, i] = input_ids[:, 0]
+
+            torch.cuda.synchronize()
+
+        return target
+
+    def inputs(self, **inputs_kwargs):
+        return {}
+
+
+class CacheSpeedBenchMark(SpeedBenchMark, CacheBenchMark):
+    pass
+
+
+if __name__ == "__main__":
+    repo_id = "meta-llama/Llama-2-7b-hf"
+    prefill_num_iter = 3
+    num_iter = 16
+
+    benchmakr = CacheSpeedBenchMark(repo_id=repo_id, prefill_num_iter=prefill_num_iter, num_iter=num_iter)
+
+    # all_batch_size = [1, 2, 4]
+    # all_max_cache_length = [4096, 2048, 1024, 512]
+    # all_seq_length = [512, 1, 1024, 2048]
+
+    all_batch_size = [1, 2]
+    all_max_cache_length = [16]
+    all_seq_length = [4]
```

Comment:
Dummy values to run and check the PR works. We should change to some meaningful values that won't OOM (depends on which machine)